test/warning.test.ts

```diff
@@ -0,0 +1,147 @@
+import { configure } from "mobx";
```

Comment:
One test I'm still missing is to verify that a client-side error (for instance if you set 'required' to true, or add in a custom validation function) overrules the getError hook derived error.